test-arquillian-container/pom.xml

```diff
@@ -0,0 +1,275 @@
+                          </address>
+                          <properties>
+                            <value>newcastle</value>
+                          </properties>
+                        </resource>
+                        <resource>
+                          <address>
+                            xa-datasource-properties=ServerName
+                          </address>
+                          <properties>
+                            <value>localhost</value>
+                          </properties>
+                        </resource>
+                        <resource>
+                          <address>
+                            xa-datasource-properties=User
+                          </address>
+                          <properties>
+                            <value>newcastle</value>
+                          </properties>
+                        </resource>
+                        <resource>
+                          <address>
+                            xa-datasource-properties=Password
+                          </address>
+                          <properties>
+                            <value>newcastle</value>
+                          </properties>
+                        </resource>
+                      </resources>
+                    </resource>
+                  </resources>
+                </configuration>
+              </execution>
+              <execution>
+                <id>shutdown-server</id>
+                <phase>generate-test-resources</phase>
+                <goals>
+                  <goal>shutdown</goal>
+                </goals>
+              </execution>
+            </executions>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
+
+    <profile>
+      <id>wildfly-820</id>
+      <activation>
+        <property>
+          <name>!eap</name>
+        </property>
+      </activation>
+      <build>
+        <plugins>
+          <plugin>
+            <artifactId>maven-dependency-plugin</artifactId>
+            <executions>
+              <execution>
+                <id>unpack-wildfly</id>
+                <phase>generate-test-resources</phase>
+                <goals>
+                  <goal>unpack</goal>
+                </goals>
+                <configuration>
+                  <artifactItems>
+                    <artifactItem>
+                      <groupId>org.wildfly</groupId>
+                      <artifactId>wildfly-dist</artifactId>
+                      <version>${jboss.version}</version>
+                      <type>zip</type>
+                      <overWrite>true</overWrite>
+                      <outputDirectory>${test.server.unpack.dir}</outputDirectory>
+                    </artifactItem>
+                  </artifactItems>
+                </configuration>
+              </execution>
+            </executions>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
+
+    <profile>
```

Comment:
All Wildfly/EAP downloading profiles are placed here. We want to download everything only once.